### PR TITLE
MKL Syrk wrapper

### PR DESCRIPTION
**Pull request by KulikovNikita**

### CODE REVIEW

**Review by Alexandr-Solovev — APPROVED** [review]
LGTM

File: cpp/oneapi/dal/backend/primitives/blas/syrk_dpc.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    ONEDAL_ASSERT(a.get_dimension(1) == c.get_dimension(0));
+    ONEDAL_ASSERT(c.get_dimension(0) == c.get_dimension(1));
+    ONEDAL_ASSERT(c.has_mutable_data());
+
+    const auto nd = c.get_dimension(0);
+    const auto kd = a.get_dimension(0);
+    const auto* const a_ptr = a.get_data();
+    auto* const c_ptr = c.get_mutable_data();
+    const auto a_str = a.get_leading_stride();
+    const auto c_str = c.get_leading_stride();
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/backend/primitives/blas/syrk_dpc.cpp:60]: This name looks like a string for me

File: cpp/oneapi/dal/backend/primitives/blas/misc.hpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
```
[the remark this answers is not in the document]
Comment (KulikovNikita) [reply cpp/oneapi/dal/backend/primitives/blas/misc.hpp:1]: It have been discussed with Ruslan some time ago. Misc seems to be a valid naming for such service functionality

File: cpp/oneapi/dal/backend/primitives/blas/syrk_dpc.cpp (hunk 1 above)
Comment (KulikovNikita) [reply cpp/oneapi/dal/backend/primitives/blas/syrk_dpc.cpp:60]: 1) As far as I know there are almost no strings in oneDAL library. Thus it seems strange to confuse stride with string in such context.
2) Unfortunately there are many places where this wording is already in use.

**Pull request closed without merging**